overlord/snapstate/snapmgr.go

```diff
@@ -58,11 +58,12 @@ func (ss *SnapSetup) MountDir() string {
 
 // SnapState holds the state for a snap installed in the system.
 type SnapState struct {
-	Sequence  []*snap.SideInfo `json:"sequence"` // Last is current
-	Candidate *snap.SideInfo   `josn:"candidate,omitempty"`
-	Active    bool             `json:"active,omitempty"`
-	Channel   string           `json:"channel,omitempty"`
-	DevMode   bool             `json:"dev-mode,omitempty"`
+	Sequence             []*snap.SideInfo `json:"sequence"` // Last is current
+	Candidate            *snap.SideInfo   `josn:"candidate,omitempty"`
+	Active               bool             `json:"active,omitempty"`
+	Channel              string           `json:"channel,omitempty"`
+	DevMode              bool             `json:"dev-mode,omitempty"`
+	LastSideloadRevision int              `json:"last-sideload-revision,omitempty"`
```

Comment:
Can we please call this "generated-revision" or similar? The term "sideload" still feels somewhat awkward to me in this context, and I suspect the introduction of assertions will blur those lines further and make it even less friendly. Over time I think the term "devmode" will take over.